waspc/data/Generator/templates/react-app/src/router/linkHelpers.ts

```diff
@@ -0,0 +1,49 @@
+export type ParamValue = string | number
+export type Params = { [name: string]: ParamValue }
+export type Search =
+  | string[][]
+  | Record<string, string>
+  | string
+  | URLSearchParams
+
+export function interpolatePath(
+  path: string,
+  params?: Params,
+  search?: Search,
+  hash?: string
+) {
+  const interpolatedPath = params ? interpolatePathParams(path, params) : path
+  const interpolatedSearch = search
+    ? `?${new URLSearchParams(search).toString()}`
+    : ''
+  const interpolatedHash = hash ? `#${hash}` : ''
+  return interpolatedPath + interpolatedSearch + interpolatedHash
+}
+
+function interpolatePathParams(path: string, params: Params) {
+  const startsWithSlash = path.startsWith("/");
+  const interpolatedPath = path
+    .split("/")
+    .map((part) => {
+      if (part.startsWith(":")) {
+        const paramName = getParamName(part);
+        return params[paramName];
+      }
+      return part;
```

Comment:
We might have misunderstood each other here. I meant extracting this part into a named function as well:
```typescript
if (part.startsWith(':')) {
    const paramName = getParamName(part)
    return params[paramName]
}
return part
```
Seems a little long for a lambda and easily nameable. If we can put a name on it, I say go for it. I know you're using a closure, but placing the definition inside the parent function should take care of that.